server/lobbyconnection.py

```diff
@@ -505,14 +515,69 @@ async def check_policy_conformity(self, player_id, uid_hash, session, ignore_res
 
         return response.get("result", "") == "honest"
 
+    async def command_auth(self, message):
+        token = message["token"]
+        unique_id = message["unique_id"]
+        player_id = await self.oauth_service.get_player_id_from_token(token)
+
+        new_irc_password = hexlify(os.urandom(16)).decode()
+        await self.send({
+            "command": "irc_password",
+            "password": new_irc_password
+        })
+
+        async with self._db.acquire() as conn:
+            result = await conn.execute(
+                select([t_login.c.login, t_login.c.steamid])
+                .where(t_login.c.id == player_id)
+            )
+            row = await result.fetchone()
+
+            username = row.login
+            steamid = row.steamid
+
+        await self.on_player_login(
+            player_id, username, new_irc_password, steamid, unique_id
+        )
+
     async def command_hello(self, message):
         login = message["login"].strip()
         password = message["password"]
+        unique_id = message["unique_id"]
 
         async with self._db.acquire() as conn:
-            player_id, login, steamid = await self.check_user_login(conn, login, password)
-            metrics.user_logins.labels("success").inc()
+            player_id, username, steamid = await self.check_user_login(
+                conn, login, password
+            )
+
+        await self.on_player_login(
+            player_id, username, password, steamid, unique_id
+        )
+
+    async def on_player_login(
+        self,
+        player_id: int,
+        username: str,
+        password: str,
+        steamid: int,
+        unique_id: str
+    ):
+        conforms_policy = await self.check_policy_conformity(
+            player_id, unique_id, self.session,
+            ignore_result=(
+                steamid is not None or
+                self.player_service.is_uniqueid_exempt(player_id)
+            )
+        )
+        if not conforms_policy:
+            return
 
+        self._logger.debug(
+            "Login from: %s, %s, %s", player_id, username, self.session
+        )
+        metrics.user_logins.labels("success").inc()
```

Comment:
Do we want to include the type of login (password or token) in the metrics at all?

Reply:
Probably would be good